src/gfx.cpp

```diff
@@ -1445,20 +1438,75 @@ void RedrawScreenRect(int left, int top, int right, int bottom)
 	VideoDriver::GetInstance()->MakeDirty(left, top, right - left, bottom - top);
 }
 
+static std::vector<Rect> _dirty_viewport_occlusions;
+static const Viewport *_dirty_viewport;
+static NWidgetDisplay _dirty_viewport_disp_flags;
```

Comment:
global variables should go together at the top of the file

Reply:
@JGRennison I hope you're taking note of this -- most of these are in your commits :wink:

Reply:
What is the policy on "global" variables that are used only by very few local, related functions; e.g. `_confirmed_screenshot_type` in `screenshot.cpp`? Obviously, the ideal in that particular situation would be to use a closure; but I don't think the codebase is ready for those yet?

Reply:
In this particular case, those three globals are used only by `DrawDirtyViewport` and `DrawDirtyBlocks`; and `DrawDirtyViewport` is only used by itself and `DrawDirtyBlocks`. It seems like it should be relatively straightforward to eliminate them.